### PR TITLE
Feature/create location

### PR DESCRIPTION
**Pull request by alexandreMelloTW**
:warning: _Is there something everyone needs to know before merging?_ (Optional)

### Requirement 
<details>
    <summary>
        <strong>Clearly explain your requirement.</strong> 
    </summary>
    <p>
       Implement an endpoint to create a Location.
    </p>
</details>

### Solution
<details>
    <summary>
        <strong>Clearly explain your solution.</strong>           
    </summary>
    <p>
1. Created `Location Controller` to receive an `Location Form` and send it to `Location Service`

2. Created `Location Service` to _map_ `Location Form` into `Location` and save it by `Location Repository`.

3. Created `Location Repository` to handle _database persistence_.
    </p>
</details>

### How to test
<details>
    <summary>
        <strong>Request for test.</strong>
    </summary>
    <p>
curl -v -X POST http://localhost:8080/location -H 'Content-Type: application/json' -d '{ "country": "Brazil", "state": "SP", "city": "São Paulo" }'
   </p>
</details>

### CODE REVIEW

**Review by MarceloLeite2604 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -85,6 +85,14 @@
             <artifactId>spring-boot-configuration-processor</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-contract-shade</artifactId>
```
Comment (fwfurtado) [inline pom.xml]: Do we need this dependency? If so, for what?

File: src/main/java/br/com/caelum/clines/shared/domain/Country.java [hunk 2]
```diff
@@ -264,4 +265,14 @@
     public String getDescription() {
         return description;
     }
+
+    public static Country findByDescription(String description) {
+        Assert.notNull(description, "Description cannot be null");
+        for (Country value : values()) {
```
Comment (fwfurtado) [inline src/main/java/br/com/caelum/clines/shared/domain/Country.java]: What do you think about to use: 
```java
EnumSet.allOf(Country.class)
.stream()
.filter(countryDescription -> countryDescription.equals(description))
.findFirst()
.orElseThrow(() -> new IllegalArgumentException("Country " + description + " not found."));
```

**Review by fwfurtado — CHANGES_REQUESTED** [review]

File: src/main/java/br/com/caelum/clines/shared/domain/Country.java (hunk 2 above)
Comment (alexandreMelloTW) [reply src/main/java/br/com/caelum/clines/shared/domain/Country.java]: solved

File: pom.xml (hunk 1 above)
Comment (alexandreMelloTW) [reply pom.xml]: removed

**Review by fwfurtado — APPROVED** [review]

**Pull request merged**